fix: Use relative position for toc indicator during scroll on mobile

MoritzWeber0 opened the pull request.

therobrob commented:
i think the small gap is present in light _and_ dark mode. We have the border around the content-element, which is transparent in light mode. I've made it visible with this lightgreen color :)

<img width="220" height="699" alt="Bildschirmfoto 2025-10-18 um 12 13 41" src="https://github.com/user-attachments/assets/05f1f106-d15d-487e-978d-84d32490c272" />

<img width="323" height="719" alt="Bildschirmfoto 2025-10-18 um 12 13 28" src="https://github.com/user-attachments/assets/8e5cb3b5-a3b1-45db-9681-3c65e35b3a94" />

MoritzWeber0 commented:
> i think the small gap is present in light _and_ dark mode. We have the border around the content-element, which is transparent in light mode. I've made it visible with this lightgreen color :)

Thanks! I removed the borders on the left & right on mobile to avoid the issue.
Also removed the mixin, it was too complex for what it was doing and replaced it with two simple CSS rules.
